invenio_rdm_records/resources/serializers/datacite/__init__.py

```diff
@@ -22,6 +29,7 @@ def __init__(self, **options):
             format_serializer_cls=JSONSerializer,
             object_schema_cls=DataCite43Schema,
             list_schema_cls=BaseListSchema,
+            processors=datacite_post_dump_processors,
```

Comment:
`processors` can be passed directly to the schema, instead of delegating that responsability to `MarshmallowSerializer`. However, that requires a bigger refactor (e.g. in `flask-resources`) 

```suggestion
            format_serializer_cls=JSONSerializer,
            object_schema_cls=DataCite43Schema(processors=datacite_post_dump_processors),
            list_schema_cls=BaseListSchema,
```